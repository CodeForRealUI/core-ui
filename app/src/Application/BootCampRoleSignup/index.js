import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./BootCampRoleSignup'),
  loading: () => null,
});
