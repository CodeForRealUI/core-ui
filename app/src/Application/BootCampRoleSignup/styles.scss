@import '~sass/variables';
@import '~sass/colors';
@import '~sass/mixins';

@include pinkGradientBackground();

.bootcamp-graduate-signup-container {
     @include center-item();

     .bootcamp-graduate-signup-box {
        text-align: center;
        padding: 40px;
        max-width: 400px;

        button:disabled {
            opacity:0.5;
        }
      
        .continue-button {
            @include button();
            margin-top:30px;
        }
    }
}