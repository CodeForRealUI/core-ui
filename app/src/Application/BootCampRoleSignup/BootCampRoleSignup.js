import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Paper, TextField, FormGroup } from 'material-ui';
// import { Link } from 'react-router-dom';

import { rolePickRequest } from '~/data/actions/rolePicker';
import './styles.scss';

class BootCampRoleSignup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      code: '',
    };
  }
  onHandleRolePickRequest = e => {
    e.preventDefault();
    const payload = {
      school_code: this.state.code,
    };
    this.props.handleRolePickRequest(payload);
  };

  onEnterCode = e => {
    const { value } = e.target;
    this.setState({
      code: value,
    });
  };

  renderForm = () => (
    <div className="bootcamp-graduate-signup-fields">
      <form onSubmit={this.onHandleRolePickRequest}>
        <FormGroup>
          <TextField
            autoFocus
            fullWidth
            placeholder="Enter your code"
            required
            onChange={this.onEnterCode}
          />
          <button
            className="continue-button"
            disabled={this.state.code.length < 24}
            onClick={this.onHandleRolePickRequest}
          >
            Continue
          </button>
        </FormGroup>
      </form>
    </div>
  );

  render() {
    return (
      <div className="bootcamp-graduate-signup-container">
        <Paper className="bootcamp-graduate-signup-box" elevation={24}>
          <h1>Bootcamp Graduate Signup</h1>
          {this.renderForm()}
        </Paper>
      </div>
    );
  }
}

BootCampRoleSignup.propTypes = {
  handleRolePickRequest: PropTypes.func.isRequired,
};

export default connect(null, dispatch => ({
  handleRolePickRequest: code => dispatch(rolePickRequest(code)),
}))(BootCampRoleSignup);
