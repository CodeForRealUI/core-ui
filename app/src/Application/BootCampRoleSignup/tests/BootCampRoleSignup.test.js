import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { BootCampRoleSignup } from '../BootCampRoleSignup';

describe('Component BootCampRoleSignup', () => {
  const handleRolePickRequest = jest.fn();
  const props = {
    handleRolePickRequest,
  };
  it('should dispatch handleRolePickRequest firing onHandleRolePickRequest with this.state.code as payload', () => {
    const wrapper = shallow(<BootCampRoleSignup {...props} />);
    expect(handleRolePickRequest.mock.calls.length).toBe(0);
    wrapper.setState({
      code: '123',
    });
    const event = {
      preventDefault() {
        return '';
      },
    };
    wrapper.instance().onHandleRolePickRequest(event);
    expect(handleRolePickRequest.mock.calls.length).toBe(1);
    expect(handleRolePickRequest.mock.calls[0][0]).toEqual({
      school_code: '123',
    });
  });
  it('should setState on firing onEnterCode with an event', () => {
    const wrapper = shallow(<BootCampRoleSignup {...props} />);
    const event = {
      target: {
        value: '123',
      },
    };
    wrapper.instance().onEnterCode(event);
    expect(wrapper.instance().state).toEqual({
      code: '123',
    });
  });
  it('should enable the button if state.code is >= 24 ', () => {
    const wrapper = shallow(<BootCampRoleSignup {...props} />);
    wrapper.setState({
      code: [].length = 24,
    });
    expect(wrapper.find('button').props().disabled).toBe(false);
  });
  it('should disable the button if state.code is < 24 ', () => {
    const wrapper = shallow(<BootCampRoleSignup {...props} />);
    wrapper.setState({
      code: '123',
    });
    expect(wrapper.find('button').props().disabled).toBe(true);
  });

  it('should fire off the onHandleRolePickRequest when button is clicked', () => {
    handleRolePickRequest.mockReset();
    const wrapper = shallow(<BootCampRoleSignup {...props} />);
    expect(handleRolePickRequest.mock.calls.length).toBe(0);
    wrapper.setState({
      code: '321',
    });
    wrapper.find('button').simulate('submit', {
      preventDefault() {
        return '';
      },
    });
    expect(handleRolePickRequest.mock.calls.length).toBe(1);
    expect(handleRolePickRequest.mock.calls[0][0]).toEqual({
      school_code: '321',
    });
  });
  it('should remain consistent if no change was intended', () => {
    const tree = renderer.create(<BootCampRoleSignup {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
