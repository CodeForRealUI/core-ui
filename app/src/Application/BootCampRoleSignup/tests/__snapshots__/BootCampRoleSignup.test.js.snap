// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component BootCampRoleSignup should remain consistent if no change was intended 1`] = `
<div
  className="bootcamp-graduate-signup-container"
>
  <div
    className="MuiPaper-root-1 MuiPaper-elevation24-27 MuiPaper-rounded-2 bootcamp-graduate-signup-box"
  >
    <h1>
      Bootcamp Graduate Signup
    </h1>
    <div
      className="bootcamp-graduate-signup-fields"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="MuiFormGroup-root-28"
        >
          <div
            aria-describedby={undefined}
            className="MuiFormControl-root-30 MuiFormControl-fullWidth-33"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              className="MuiInput-root-34 MuiInput-fullWidth-41 MuiInput-formControl-35 MuiInput-underline-38"
            >
              <input
                aria-invalid={false}
                aria-required={true}
                autoComplete={undefined}
                autoFocus={true}
                className="MuiInput-input-42"
                defaultValue={undefined}
                disabled={false}
                id={undefined}
                name={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder="Enter your code"
                readOnly={undefined}
                required={true}
                rows={undefined}
                type="text"
                value={undefined}
              />
            </div>
          </div>
          <button
            className="continue-button"
            disabled={true}
            onSubmit={[Function]}
          >
            Continue
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
